File: 01-09-2024_convert 1D array to 2D array.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> create2DArray(vector<int>& arr, int rows, int cols) {
        if (rows * cols != arr.size()) {
            return vector<vector<int>>(0, vector<int>(0));
        }
        vector<vector<int>> matrix(rows, vector<int>(cols));
        int idx = 0;
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                matrix[i][j] = arr[idx];
                idx++;
            }
        }
        return matrix;
    }
};

File: 01-09-2024_convert 1D array to 2D array.py
class Solution:
    def create2DArray(self, arr, rows, cols):
        if rows * cols != len(arr):
            return []
        matrix = [[0] * cols for _ in range(rows)]
        idx = 0
        for i in range(rows):
            for j in range(cols):
                matrix[i][j] = arr[idx]
                idx += 1
        return matrix
